airflow/providers/google/cloud/hooks/cloud_build.py

```diff
@@ -69,82 +67,568 @@ def __init__(
+        :type project_id: Optional[str]
+        :param retry: Optional, a retry object used  to retry requests. If `None` is specified, requests
+            will not be retried.
+        :type retry: Optional[Retry]
+        :param timeout: Optional, the amount of time, in seconds, to wait for the request to complete.
+            Note that if `retry` is specified, the timeout applies to each individual attempt.
+        :type timeout: Optional[float]
+        :param metadata: Optional, additional metadata that is provided to the method.
+        :type metadata: Optional[Sequence[Tuple[str, str]]]
+
+        :rtype: `google.cloud.devtools.cloudbuild_v1.types.BuildTrigger`
+        """
+        if not project_id:
+            raise ValueError("The project_id should be set")
+
+        client = self.get_conn()
+        return client.update_build_trigger(
+            project_id=project_id,
+            trigger_id=trigger_id,
+            trigger=trigger,
+            retry=retry,
+            timeout=timeout,
+            metadata=metadata,
+        )
+
+    def _wait_for_operation_to_complete(
+        self, func: Callable, **kwargs: Optional[dict]
+    ) -> dict:
         """
         Waits for the named operation to complete - checks status of the
         asynchronous call.
 
-        :param operation_name: The name of the operation.
-        :type operation_name: str
+        :param func: The function that needs to be called.
+        :type func: Callable
+        :param kwargs: dict of function keyword arguments
+        :type kwargs: dict
         :return: The response returned by the operation.
         :rtype: dict
         :exception: AirflowException in case error is returned.
         """
-        service = self.get_conn()
         while True:
-            operation_response = (
-                # pylint: disable=no-member
-                service.operations().get(name=operation_name).execute(num_retries=self.num_retries)
-            )
-            if operation_response.get("done"):
-                response = operation_response.get("response")
-                error = operation_response.get("error")
-                # Note, according to documentation always either response or error is
-                # set when "done" == True
-                if error:
-                    raise AirflowException(str(error))
-                return response
+            operation = func(**kwargs)
+            operation_dict = MessageToDict(operation)
+            status = operation_dict["status"]
+            if status:
+                if status == "SUCCESS":
+                    return operation_dict
+                elif status in [
+                    "FAILURE",
+                    "INTERNAL_ERROR",
+                    "TIMEOUT",
+                    "CANCELLED",
+                    "EXPIRED",
+                ]:
+                    raise AirflowException(str(operation_dict))
```

Comment:
Won't `operation.result()` do the same trick as `while` loop?

Reply:
Yeah, you are right. I think they have fixed the issue by replacing `_OperationFuture` with `Operation`. Now we can use that to check the status.